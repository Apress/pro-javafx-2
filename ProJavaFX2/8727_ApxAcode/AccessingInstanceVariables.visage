/*
 * @test
 * @run
 */

class Point {
  var x: Number;
  var y: Number;
  override function toString(): String {
    "Point({x}, {y})"
  }
}

// reading member variables
var p = Point { x: 3.0, y: 4.0 };
println("p.x = {p.x}");
println("p.y = {p.y}");
println("p = {p}");

// writing to member variables
p.x = 5.0;
p.y = 12.0;
println("p = {p}");
