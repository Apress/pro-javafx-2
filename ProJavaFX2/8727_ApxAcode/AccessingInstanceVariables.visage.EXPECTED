p.x = 3.0
p.y = 4.0
p = Point(3.0, 4.0)
p = Point(5.0, 12.0)
